File: src/main/java/pl/denisolek/User/UserService.java
package pl.denisolek.User;

import org.springframework.http.HttpStatus;
import org.springframework.stereotype.Component;
import pl.denisolek.Exception.ServiceException;
import pl.denisolek.Reservation.Reservation;
import pl.denisolek.Restaurant.Restaurant;
import pl.denisolek.User.Requests.UserRegistrationRequest;

import java.util.List;

@Component
public class UserService {

    private final UserRepository userRepository;

    public UserService(UserRepository userRepository) {
        this.userRepository = userRepository;
    }

    public User addUser(UserRegistrationRequest userRegistrationRequest) {
        User newUser = new User();
        Restaurant newRestaurant = new Restaurant();

        newRestaurant.setName(userRegistrationRequest.getRestaurantName());
        newRestaurant.setCity(userRegistrationRequest.getRestaurantCity());
        newRestaurant.setStreet(userRegistrationRequest.getRestaurantStreet());
        newRestaurant.setLatitude(userRegistrationRequest.getRestaurantLatitude());
        newRestaurant.setLongitude(userRegistrationRequest.getRestaurantLongitude());
        newRestaurant.setAvgReservationTime(userRegistrationRequest.getRestaurantAvgReservationTime());
        newRestaurant.setNip(userRegistrationRequest.getRestaurantNip());
        newRestaurant.setCapacity(userRegistrationRequest.getRestaurantCapacity());

        newUser.setEmail(userRegistrationRequest.getEmail());
        newUser.setName(userRegistrationRequest.getName());
        newUser.setSurname(userRegistrationRequest.getSurname());
        newUser.setRestaurant(newRestaurant);

        return userRepository.save(newUser);
    }

    public Restaurant getUserRestaurant(User user) {
        if (user == null)
            throw new ServiceException(HttpStatus.NOT_FOUND, "User not found");

        return user.getRestaurant();
    }

    public List<Reservation> getUserRestaurantReservations(User user) {
        if (user == null)
            throw new ServiceException(HttpStatus.NOT_FOUND, "User not found");

        return user.getRestaurant().getReservations();
    }
}

File: src/main/kotlin/pl/denisolek/core/user/UserService.kt
package pl.denisolek.core.user

import org.springframework.stereotype.Service
import pl.denisolek.core.restaurant.Restaurant
import pl.denisolek.infrastructure.util.generateUsernameString

@Service
class UserService(val userRepository: UserRepository) {
    fun save(user: User) =
            userRepository.save(user)

    fun findByEmail(email: String): User? =
            userRepository.findByEmail(email)

    fun findByUsername(username: String): User? =
            userRepository.findByUsername(username)

    fun findByRestaurant(restaurant: Restaurant): User? =
            userRepository.findByRestaurant(restaurant)

    fun generateUsername(): String {
        var exists: Boolean = true
        var username: String = "username"

        while (exists) {
            username = generateUsernameString()
            if (userRepository.countByUsername(username) == 0)
                exists = false
        }
        return username
    }
}